### ISSUE TITLE
.travis failing on compile with boost and google test

### ISSUE THREAD

**Issue by philass** [sorenlassen/8heap#2]
Travis build fails with 
```
0.90s$ make HeapAuxTest
Scanning dependencies of target HeapAuxTest
[ 50%] Building CXX object CMakeFiles/HeapAuxTest.dir/HeapAuxTest.cpp.o
/home/travis/build/sorenlassen/8heap/HeapAuxTest.cpp:18:25: error: ‘boost::iterators::counting_iterator’ has not been declared
 using boost::iterators::counting_iterator;
                         ^~~~~~~~~~~~~~~~~
/home/travis/build/sorenlassen/8heap/HeapAuxTest.cpp:19:25: error: ‘boost::iterators::transform_iterator’ has not been declared
 using boost::iterators::transform_iterator;
                         ^~~~~~~~~~~~~~~~~~
/home/travis/build/sorenlassen/8heap/HeapAuxTest.cpp:32:17: error: expected constructor, destructor, or type conversion before ‘(’ token
 TYPED_TEST_SUITE(HeapAuxTest, Implementations);
                 ^
In file included from /usr/include/gtest/gtest.h:66:0,
                 from /home/travis/build/sorenlassen/8heap/HeapAuxTest.cpp:14:
/home/travis/build/sorenlassen/8heap/HeapAuxTest.cpp:34:1: error: ‘gtest_type_params_HeapAuxTest_’ was not declared in this scope
 TYPED_TEST(HeapAuxTest, Clear) {
 ^
/home/travis/build/sorenlassen/8heap/HeapAuxTest.cpp:34:1: error: template argument 3 is invalid
 TYPED_TEST(HeapAuxTest, Clear) {
 ^
/home/travis/build/sorenlassen/8heap/HeapAuxTest.cpp:42:1: error: ‘gtest_type_params_HeapAuxTest_m’ was not declared in this scope
 TYPED_TEST(HeapAuxTest, Push3) {
 ^
/home/travis/build/sorenlassen/8heap/HeapAuxTest.cpp:42:1: error: template argument 3 is invalid
 TYPED_TEST(HeapAuxTest, Push3) {
 ^
/home/travis/build/sorenlassen/8heap/HeapAuxTest.cpp:64:1: error: ‘gtest_type_params_HeapAuxTest_’ was not declared in this scope
 TYPED_TEST(HeapAuxTest, Heapify3) {
 ^
/home/travis/build/sorenlassen/8heap/HeapAuxTest.cpp:64:1: error: template argument 3 is invalid
 TYPED_TEST(HeapAuxTest, Heapify3) {
 ^
/home/travis/build/sorenlassen/8heap/HeapAuxTest.cpp:77:1: error: ‘gtest_type_params_HeapAuxTest_’ was not declared in this scope
 TYPED_TEST(HeapAuxTest, Sort3) {
 ^
/home/travis/build/sorenlassen/8heap/HeapAuxTest.cpp:77:1: error: template argument 3 is invalid
 TYPED_TEST(HeapAuxTest, Sort3) {
 ^
/home/travis/build/sorenlassen/8heap/HeapAuxTest.cpp:91:1: error: ‘gtest_type_params_HeapAuxTest_’ was not declared in this scope
 TYPED_TEST(HeapAuxTest, Heapify100) {
 ^
/home/travis/build/sorenlassen/8heap/HeapAuxTest.cpp:91:1: error: template argument 3 is invalid
 TYPED_TEST(HeapAuxTest, Heapify100) {
 ^
/home/travis/build/sorenlassen/8heap/HeapAuxTest.cpp: In member function ‘virtual void {anonymous}::HeapAuxTest_Heapify100_Test<gtest_TypeParam_>::TestBody()’:
/home/travis/build/sorenlassen/8heap/HeapAuxTest.cpp:95:3: error: ‘counting_iterator’ was not declared in this scope
   counting_iterator<value_type> zero(0);
   ^~~~~~~~~~~~~~~~~
/home/travis/build/sorenlassen/8heap/HeapAuxTest.cpp:95:3: note: suggested alternative:
In file included from /home/travis/build/sorenlassen/8heap/HeapAuxTest.cpp:12:0:
/usr/include/boost/iterator/counting_iterator.hpp:22:7: note:   ‘boost::counting_iterator’
 class counting_iterator;
       ^~~~~~~~~~~~~~~~~
/home/travis/build/sorenlassen/8heap/HeapAuxTest.cpp:95:31: error: expected primary-expression before ‘>’ token
   counting_iterator<value_type> zero(0);
                               ^
/home/travis/build/sorenlassen/8heap/HeapAuxTest.cpp:95:39: error: there are no arguments to ‘zero’ that depend on a template parameter, so a declaration of ‘zero’ must be available [-fpermissive]
   counting_iterator<value_type> zero(0);
                                       ^
/home/travis/build/sorenlassen/8heap/HeapAuxTest.cpp:95:39: note: (if you use ‘-fpermissive’, G++ will accept your code, but allowing the use of an undeclared name is deprecated)
/home/travis/build/sorenlassen/8heap/HeapAuxTest.cpp:100:35: error: ‘zero’ was not declared in this scope
   auto begin = transform_iterator(zero, revert);
                                   ^~~~
make[3]: *** [CMakeFiles/HeapAuxTest.dir/HeapAuxTest.cpp.o] Error 1
make[2]: *** [CMakeFiles/HeapAuxTest.dir/all] Error 2
make[1]: *** [CMakeFiles/HeapAuxTest.dir/rule] Error 2
make: *** [HeapAuxTest] Error 2
The command "make HeapAuxTest" exited with 2.
0.00s$ ./HeapAuxTest
/home/travis/.travis/functions: line 109: ./HeapAuxTest: No such file or directory
The command "./HeapAuxTest" exited with 127.
```
My guess is this is due to wrong Boost and Google Test versions being installed in the .travis build

**Comment by philiplassen** [comment]
Commit had wrong issue number. This is yet to be fixed.

**Comment by philass** [comment]
Fixed in #8
